### PR TITLE
Update Resources tutorial to mention and use the GlobalClass annotation for C#

### PR DESCRIPTION
**Pull request by j4d3blooded**
The previous version of this page didn't mention using the `[GlobalClass]` annotation. Following this tutorial results in the user not finding their resource in the Create a new resource menu without any understanding as to why

### CODE REVIEW

File: tutorials/scripting/resources.rst [hunk 1]
```diff
@@ -208,6 +208,9 @@ It should appear in your file tab with the full name ``bot_stats.tres``.
 Without a script, it's useless, so let's add some data and logic!
 Attach a script to it named ``bot_stats.gd`` (or just create a new script, and then drag it to it).
 
+.. note::
+    If you're using C#, you need to annoate your Resource class with the ``[GlobalClass]`` annotation for it to show up in the create resource GUI.
```
Comment (paulloz) [inline tutorials/scripting/resources.rst:212]:
```suggestion
    If you're using C#, you need to annotate your Resource class with the ``[GlobalClass]`` attribute for it to show up in the create resource GUI.
```

This is called an attribute in C#, not an annotation.

**Review by paulloz — CHANGES_REQUESTED** [review]
Thank you!

**Review by paulloz — APPROVED** [review]

File: tutorials/scripting/resources.rst [hunk 2]
```diff
@@ -329,7 +333,7 @@ Now, select the :ref:`CharacterBody3D <class_CharacterBody3D>` node which we nam
       .. code-tab:: csharp
 
         using Godot;
-
+        [GlobalClass]
```
Comment (AThousandShips) [inline tutorials/scripting/resources.rst:336]:
```suggestion

        [GlobalClass]
```

**Review by AThousandShips — CHANGES_REQUESTED** [review]
Thank you for your contribution!

You should target `master` not `4.2`, changes can then be cherry picked, see [here](https://docs.godotengine.org/en/latest/contributing/documentation/contributing_to_the_documentation.html#contributing-changes).

You can rebase it to fix it, or open a new one, see [here](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#rebasing-onto-another-branch)

**Comment by j4d3blooded** [comment]
I'm not entirely sure how I would go about fixing merge commits or cherry picking the relevant changes from this point, sorry

**Review by AThousandShips — APPROVED** [review]

**Comment by AThousandShips** [comment]
There, it works now, just had to change the target as well, thank you!

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Merged. Thanks and congrats on your first merged contribution!

**Pull request merged**